app/controllers/profile_controller.rb

```diff
@@ -11,4 +11,45 @@ def index
     @stories = Kaminari.paginate_array(get_stories(user))
                        .page(params[:page])
   end
+
+  def add_ban
+    ban(true)
+  end
+
+  def remove_ban
+    ban(false)
+  end
+
+  private
+
+  def ban_mailer(user, banned)
+    return unless user.any?
+
+    if banned
+      BannedMailer.add_ban_email(user.first).deliver_now
```

Comment:
Do we really need to pass the whole user object to the mailer method?

AFAIK, the arguments passed to the mailer method will be serialized and stored in the queue to be processed later. Serializing a heavy `User` object is expensive and we should avoid it . Someone let me know if I'm wrong.

I suggest we pass the `user_id` to the mailer method and let it do the job of looking up the `User` object.

@julianguyen  What do you think?

Reply:
That's a good suggestion, will update :D Thanks!